### PR TITLE
Spark impersonation

### PR DESCRIPTION
**Pull request by hayssams**
### What is this PR for?
Makes Spark Interperter impersonate front-end user using spark.yarn.principal who becomes the only user to be declared in hadoop core-site.xml (hadoop.proxyuser.USER.hosts)
Added Kerberos auto-renewal flag in zeppelin-site.xml (zeppelin.kerberos.renew.period) / ZEPPELIN_KERBEROS_RENEW_PERIOD env var

### What type of PR is it?
[Bug Fix]

### Todos
* [ ] - Task

### What is the Jira issue?
* [ZEPPELIN-2066]

### How should this be tested?
* Manually against kerberized HDFS cluster.

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes. To document the kerberos ticket renewal period flag

### CODE REVIEW

File: zeppelin-web/src/app/interpreter/interpreter-create.html [hunk 1]
```diff
@@ -196,7 +196,7 @@ <h5>Option</h5>
           </div>
         </div>
         <div class="row interpreter" style="margin-top: 5px;"
-             ng-show="getInterpreterRunningOption(setting.id)=='Per User' && getPerUserOption(setting.id)=='isolated'">
+             ng-show="(getInterpreterRunningOption(setting.id)=='Per Note' || getInterpreterRunningOption(setting.id)=='Per User') && getPerUserOption(setting.id)=='isolated'">
```
Comment (hayssams) [inline zeppelin-web/src/app/interpreter/interpreter-create.html]: reverted this one.

File: zeppelin-web/src/app/interpreter/interpreter.html [hunk 2]
```diff
@@ -315,7 +315,7 @@ <h5>Option</h5>
     </div>
 
     <div class="row interpreter" style="margin-top: 5px;"
-         ng-show="getInterpreterRunningOption(setting.id)=='Per User' && getPerUserOption(setting.id)=='isolated'">
+         ng-show="(getInterpreterRunningOption(setting.id)=='Per Note' || getInterpreterRunningOption(setting.id)=='Per User') && getPerUserOption(setting.id)=='isolated'">
```
Comment (hayssams) [inline zeppelin-web/src/app/interpreter/interpreter.html]: reverted this one.

**Comment by zjffdu** [comment]
Thanks @hayssams for the contribution. Several comments 
1. Have you verified the current mechanism of kerberos support ? 
2. This PR seems only work with yarn client mode, but not yarn cluster mode. As in yarn cluster mode, there's no keytab file in driver side.

**Comment by hayssams** [comment]
@zjffdu my bas. Ticket renewal is not required. Thanks for your clarification.

**Comment by zjffdu** [comment]
Thanks @hayssams for confirm

**Pull request closed without merging**